app/plugin.go

```diff
@@ -182,6 +189,78 @@ func (a *App) InitPlugins(pluginDir, webappPluginDir string) {
 	a.SyncPluginsActiveState()
 }
 
+// SynchPlugins will synchronize the plugins installed locally
+// with the plugin bundles available in the file store.
+func (a *App) SynchPlugins() *model.AppError {
+	mlog.Info("Synching plugins with the file store ...")
+
+	pluginDirs, err := ioutil.ReadDir(*a.Config().PluginSettings.Directory)
+	if err != nil {
+		return model.NewAppError("SynchPlugins", "app.plugin.synch.read_local_folder.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Remove plugins locally to synch with file store.
+	for _, dir := range pluginDirs {
+		if !dir.IsDir() {
+			continue
+		}
+		dirFullPath := filepath.Join(*a.Config().PluginSettings.Directory, dir.Name())
+		files, readErr := ioutil.ReadDir(dirFullPath)
```

Comment:
Suggest we just `os.Stat` to decide if the plugin is managed:
```go
var managed bool
filestoreManagedFile, err := os.Stat(filepath.Join(*a.Config().PluginSettings.Directory, dir.Name(), '.filestore'))

if os.IsNotExists(err) {
    managed = false
} else if err != nil {
    mlog.Error(...)
    continue
} else {
    managed = true
}
```

Reply:
Thanks! Wasn't familiar with `os.Stat`. It reduced a lot of code just for looking for that flag 👌